src/dataset-builder/CohortEditor.ts

```diff
@@ -180,6 +180,7 @@ export async function criteriaFromOption(
       const rangeOption = generatedOptions;
       return {
         kind: 'range',
+        id: rangeOption.id,
         rangeOption,
         name: rangeOption.name,
```

Comment:
If this was java OOP then we wouldn't need or want to copy `rangeOption` data into the criteria since it already contains `rangeOption`. Is there a good way to do this in typescript?

Reply:
So, it kinda would work, but it leads to some duplicated code under https://github.com/DataBiosphere/terra-ui/pull/4655/files#diff-51f35ba92520217d1b007a325c2bc8050bb7eb5e749afcafd5d0c2fe1fe6dd93R192, since we would then need to handle rangeOption, listOption, and domainOption differently. We could convert them all to `option` and lift it a level up, but `domainOption` doesn't seem to have a `name` field, just a `category`. Is that the same?

Reply:
Going to propose another solution for this in a follow up PR